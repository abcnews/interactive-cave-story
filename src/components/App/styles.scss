.scrollyteller {
  background-color: #1c2121 !important;
}

.panel {
  --color-panel-background: rgba(0, 0, 0, 0.8);
}

@media (min-width: 1400px) {
  .panel {
    --color-panel-background: transparent;
    margin-right: 10%;
  }
}

.panel:last-of-type {
  margin-bottom: 100vh !important;
}

.panel img {
  display: block;
  margin: 0 auto;
  max-width: 90% !important;
}
