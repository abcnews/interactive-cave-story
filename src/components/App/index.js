import Scrollyteller from '@abcnews/scrollyteller';
import React from 'react';
import Cave from '../Cave';
import styles from './styles.scss';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.onMarker = this.onMarker.bind(this);

    this.state = {
      config: {}
    };
  }

  onMarker(marker) {
    let config = {};

    if (typeof marker.shift !== 'undefined') {
      config.shift = marker.shift;
    }

    if (typeof marker.x !== 'undefined') {
      config.x = marker.x;
    }

    if (typeof marker.y !== 'undefined') {
      config.y = marker.y;
    }

    this.setState(() => ({ config }));
  }

  render() {
    const { panels } = this.props;

    return (
      <Scrollyteller
        panels={panels}
        // className={`Block is-piecemeal has-right ${styles.scrollyteller}`}
        className={styles.scrollyteller}
        // panelClassName={`Block-content u-richtext-invert is-right ${styles.panel}`}
        panelClassName={styles.panel}
        onMarker={this.onMarker}
      >
        <Cave config={this.state.config} />
      </Scrollyteller>
    );
  }
}

export default App;
