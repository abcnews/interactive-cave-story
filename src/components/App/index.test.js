import React from 'react';
import { create } from 'react-test-renderer';
import App from '.';

describe('App', () => {
  test('It renders', () => {
    const component = create(<App panels={[]} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
