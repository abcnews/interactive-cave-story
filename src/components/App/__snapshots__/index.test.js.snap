// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App It renders 1`] = `
<div
  className="index-module_base__2Ie5h "
>
  <div
    className="index-module_graphic__QVyjs index-module_before__1gOjV"
  >
    <div>
      <div
        style={
          Object {
            "height": 2041.1733333333334,
            "transform": "translateX(-256px) scale(0.3) translateY(NaNpx)",
            "transformOrigin": "50% 0px",
            "width": 1024,
          }
        }
      >
        <img
          src="cave.png"
          style={
            Object {
              "height": 2041.1733333333334,
              "maxWidth": "1000%",
              "width": 1024,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
